### PR TITLE
A simple test harness for the merge interop M1 milestone

### PR DESCRIPTION
**Pull request by zah**
This aims to enable connecting nimbus-eth2 to nimbus-eth1 within
the testing protocol described here:

https://github.com/status-im/nimbus-eth2/blob/amphora-merge-interop/docs/interop_merge.md

To execute the work-in-progress test, please run:

In terminal 1:
tests/amphora/launch-nimbus.sh

In terminal 2:
tests/amphora/check-merge-test-vectors.sh

### CODE REVIEW

**Comment by zah** [comment]
At the moment, the test fails with:

```
engine_preparePayload response: {"jsonrpc":"2.0","id":67,"error":{"code":-32000,"message":"engine_preparePayload raised an exception","data":"error sealing block header: unauthorized signer"}}
```

File: tests/amphora/amphora-interop-genesis-m1.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "config": {
+    "chainId": 1,
```
Comment (jangko) [inline tests/amphora/amphora-interop-genesis-m1.json:3]: I assume you want run nimbus in clique mode, but at the same time the `chainId` is `1`, it won't work if you left the `--network` command line param empty.

Empty `--network` will be replaced by `chainId` from custom network file, and `1` will load `MainNet` genesis, not your custom network genesis. That's why the error message is `unauthorized signer`.

If your test network doesn't care about network id, then you can use `--network:0` or something else beside `1,3,4,5, or 42`.

I know the `--network` description is incomplete, need fix.

**Comment by jangko** [comment]
right, I forget about that, 

https://github.com/status-im/nimbus-eth1/blob/c62905db2599947c26146b65728a47179ff9e038/nimbus/sealer.nim#L135

this line above maybe should've moved somewhere else, e.g. SignerEngineRef initialization/creation.

**Comment by zah** [comment]
Closing in favor of https://github.com/status-im/nimbus-eth1/pull/940

**Pull request closed without merging**